src/v0/destinations/klaviyo/util.js

```diff
@@ -47,18 +47,27 @@ const getIdFromNewOrExistingProfile = async (endpoint, payload, requestOptions)
     profileId = resp.response?.data?.id;
   } else if (resp.status === 409) {
     const { errors } = resp.response;
-    profileId = errors[0]?.meta?.duplicate_profile_id;
-  } else {
-    throw new NetworkError(
-      `Failed to create user due to ${JSON.stringify(resp.response)}`,
-      resp.status,
-      {
-        [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(resp.status),
-      },
-      `${JSON.stringify(resp.response)}`,
-    );
+    profileId = errors?.[0]?.meta?.duplicate_profile_id;
+  }
+
+  if (profileId) {
+    return profileId;
+  }
+
+  let statusCode = resp.status;
+  if (resp.status === 201 || resp.status === 409) {
```

Comment:
If status is 201 or 409 why are we retrying?

Reply:
If we are getting 409, but in response not getting the profileId. We are retrying in that case and those events get successful after retry.

Reply:
Yes, for 409 it is getting is succeeded in retry